pkg/test/proxy/envoy/envoy.go

```diff
@@ -104,10 +142,32 @@ func (e *Envoy) Start() error {
 	e.cmd = exec.Command(envoyPath, args...)
 	e.cmd.Stderr = os.Stderr
 	e.cmd.Stdout = os.Stdout
-	if err := e.cmd.Start(); err != nil {
-		e.returnBaseID()
+	if err = e.cmd.Start(); err != nil {
 		return err
 	}
+
+	endTime := time.Now().Add(startTimeout)
+	for {
+		// Sleep a short before attempting to connect to Envoy.
+		time.Sleep(1 * time.Second)
```

Comment:
Why do we need this sleep? To let Envoy state to quiesce? 1 second granularity seems to high (I've been experimenting with sleep/retry loops when doing the Docker based model, and I've found that 1 second delay to be a significant source of startup delay. 

Instead, can we do a "reasonable" initial delay, coupled with a short retry delay? How quick does the GetServerInfo return if there is an error?